adapter/serviceControl/serviceController.go

```diff
@@ -0,0 +1,197 @@
+			name,
+			"Writes log entries to GCP service controller",
+			new(config.Params),
+		), new(clientImpl)})
+}
+
+func (*builder) ValidateConfig(c adapter.Config) (ce *adapter.ConfigErrors) {
+	return nil
+}
+
+func (b *builder) NewMetricsAspect(env adapter.Env, cfg adapter.Config, metrics map[string]*adapter.MetricDefinition) (adapter.MetricsAspect, error) {
+	return b.newAspect(env, cfg)
+}
+
+func (b *builder) NewApplicationLogsAspect(env adapter.Env, cfg adapter.Config) (adapter.ApplicationLogsAspect, error) {
+	return b.newAspect(env, cfg)
+}
+
+func (*builder) NewAccessLogsAspect(env adapter.Env, cfg adapter.Config) (adapter.AccessLogsAspect, error) {
+	return nil, errors.New("access logs are not supported")
+}
+
+func (b *builder) newAspect(env adapter.Env, cfg adapter.Config) (*aspect, error) {
+	params := cfg.(*config.Params)
+	ss, err := b.client.create(env.Logger())
+	if err != nil {
+		return nil, err
+	}
+	return &aspect{params.ServiceName, ss, env.Logger()}, nil
+}
+
+func (a *aspect) Record(values []adapter.Value) error {
+	rq := a.record(values, time.Now(), fmt.Sprintf("mixer-metric-report-id-%d", uuid.New()))
+	rp, err := a.service.Services.Report(a.serviceName, rq).Do()
+
+	if a.logger.VerbosityLevel(4) {
+		a.logger.Infof("service control metric report operation id %s\nmetrics %v\nresponse %v",
+			rq.Operations[0].OperationId, values, rp)
+	}
+	return err
+}
+
+func (a *aspect) record(values []adapter.Value, now time.Time, operationID string) *servicecontrol.ReportRequest {
+	var vs []*servicecontrol.MetricValueSet
+	for _, v := range values {
+		// TODO Only support request count.
+		if v.Definition.Name != "request_count" {
+			a.logger.Warningf("service control metric got unsupported metric name: %s\n", v.Definition.Name)
+			continue
+		}
+		var mv servicecontrol.MetricValue
+		mv.Labels, _ = translate(v.Labels)
+		mv.StartTime = v.StartTime.Format(time.RFC3339Nano)
+		mv.EndTime = v.EndTime.Format(time.RFC3339Nano)
+		i, _ := v.Int64()
+		mv.Int64Value = &i
+
+		ms := &servicecontrol.MetricValueSet{
+			MetricName:   "serviceruntime.googleapis.com/api/producer/request_count",
+			MetricValues: []*servicecontrol.MetricValue{&mv},
+		}
+		vs = append(vs, ms)
+	}
+
+	op := &servicecontrol.Operation{
+		OperationId:     operationID,
+		OperationName:   "reportMetrics",
+		StartTime:       now.Format(time.RFC3339Nano),
+		EndTime:         now.Format(time.RFC3339Nano),
+		MetricValueSets: vs,
+		Labels: map[string]string{
+			"cloud.googleapis.com/location": "global",
+		},
+	}
+	return &servicecontrol.ReportRequest{
+		Operations: []*servicecontrol.Operation{op},
+	}
+}
+
+// translate mixer label key to GCP label, and find the api method. TODO Need general redesign.
+// Current implementation is just adding a prefix / which does not work for some labels.
+// Need to support api_method
+// "serviceruntime.googleapis.com/api_method": method,
+func translate(labels map[string]interface{}) (map[string]string, string) {
```

Comment:
same here. This only works very limited particular case. We need a whole design of configuration to handle this label translation properly